ecgtools/core.py

```diff
@@ -168,24 +171,24 @@ def get_filelist_from_dirs(
 
         if dirs is None:
             if not self.dirs:
-                self.get_directories(depth=depth)
+                self._get_directories(depth=depth)
                 dirs = self.dirs.copy()
             else:
                 dirs = self.dirs.copy()
         if lazy:
             filelist = [
-                self.get_filelist_delayed(directory, extension, exclude_patterns)
+                self._get_filelist_delayed(directory, extension, exclude_patterns)
                 for directory in dirs
             ]
         else:
             filelist = [
-                self.get_filelist(directory, extension, exclude_patterns) for directory in dirs
+                self._get_filelist(directory, extension, exclude_patterns) for directory in dirs
             ]
         self.filelist = filelist
         return self
 
     def parse_files_attributes(
```

Comment:
Specifically on the `parse_files_attributes` method...

This seems like a helper function.  The usefulness to the user (see previous comment) seems to be the creation of the `entries` attribute.  But, in its current form, it seems like `entries` is less useful than `df` which is just the `entries` attribute converted to a DataFrame.  Correct?  Maybe, if this function should remain public, the end product of this function is the generation of the `df` attribute, and then make `entries` private.  If so, then maybe the function should change its name to reflect that.

Also, this method calls a public function of the same name that exists at module scope (below).  This seems very confusing.  If the two functions did exactly the same thing, then perhaps it makes sense for the functions to have the same name, but they do not do the same thing, exactly.

So, I might pose the following solutions:

- Is the module-scope public function `parse_files_attributes` really something you want in the public API?  Or is it really more of a helper function?  If so, maybe make it private.
- If `Builder.parse_files_attributes` is most useful as a step to constructing the DataFrame, then maybe change the function's name to match that capability (like `Builder.create_dataframe`)?

Reply:
> But, in its current form, it seems like entries is less useful than df which is just the entries attribute converted to a DataFrame. Correct? 

I concur. `entries` is less useful than df. The most important pieces here are the `filelist` and the `df`. 

> Is the module-scope public function parse_files_attributes really something you want in the public API? Or is it really more of a helper function? If so, maybe make it private.

👍Will make it private, and move its arguments into `__init__()`